rfps/rfp-consensus-visualization-tool.md

```diff
@@ -0,0 +1,123 @@
+>
+> Blocks from a given round are assembled into Tipsets according to certain rules (they must share the same parents and have been mined at the same height). Different miners may mine on different Tipsets because of network propagation delay.
+>
+> [More on Tipsets in the Spec...](https://filecoin-project.github.io/specs/#tipsets)
+
+Expected Consensus based on Storage Power Mining is a very exciting consensus algorithm in Filecoin. We would like to educate those new to Filecoin on how it works. We are also interested in observing when bad forks happen and probe when and why they may be occuring.
+
+#### Related Tools
+
+- A great example of an interactive tool that shows what is happening on the Ethereum chain is [https//ethviewer.live](http://ethviewer.live/).
+- A cool live blockchain sonification and visualization tool from Aleth.io can be seen at [https://audiokitpro.com/the-sound-of-blockchain](https://audiokitpro.com/the-sound-of-blockchain/).
+- There are also explorers that show information about uncles and chain reorgs such as [https://etherscan.io/blocks_forked](https://etherscan.io/blocks_forked).
+
+> (If you have other ideas for visualizations or tools that you think would be helpful for observing Filecoin consensus let us know!)
+
+## Deliverables
+
+An open-sourced, dual-licensed (under MIT and APACHE2) implementation that:
+
+- provides an interactive visualization of Filecoin consensus viewable in web browsers
+- or alternatively, a proposal to extend and enhance the existing explorer at [http://user.kittyhawk.wtf:8000](http://user.kittyhawk.wtf:8000/) ([Github repo](https://github.com/filecoin-project/filecoin-explorer))
+- displays new messages, blocks, tipsets, ancestors, forks, miners and related information via live updates
+- visualizes what is happening as messages are pooled into blocks forming Tipsets that extend Parent and GrandParent Tipsets as consensus is formed or forks occur
+- shows when natural and bad forks occur
+- provides explainers for what is happening and a glossary of terms
+
+
+And also:
+
+- has a well-documented, human-readable codebase
+- works with a Filecoin node API (local or remote) and database or cache if needed to perform the intended workflow
+- has tutorials with demo usage
+- can be in any mainstream language with a preference for ones that ensure easy long-term maintenance
+- is well-tested and usable by a large audience during Testnet and Mainnet
+
+> NOTE: Node API improvements are likely between now, Testnet and Mainnet. We will work with node implementers to document relevant API calls and provide coordinated opportunities for questions and feedback. A filecoin-api-client npm library also exists but is currently out of date.
```

Comment:
Let's direct people to the lotus API, and definitely not the filecoin-api-client. That's currently only functional with go-filecoin (and go-filecoin has to implement much of the same API as lotus anyways), so expect a lot of that to change.